content/tutorial-stochastic-simulations.md

```diff
@@ -0,0 +1,483 @@
+does to the right.
+At the bottom of the board, the marbles collect in bins with the height of
+marbles yielding a normal distribution.
+
++++
+
+![](tutorial-stochastic/galton_board.png)
+
++++
+
+**Figure 1.** An example of a Galton board with three marbles with three
+different potential paths (dotted lines) shown.
+
+The key operational detail of a Galton board is that every marble moves one
+increment to the left or right at each level, so our simulation centers around
+a series of randomly generated +1 and -1 to determine this movement by the
+following methodolog.
+1. Generate a random series of +1 and -1 in an even distribution - one value for
+   every level on the Galton board
+2. Sum up the values in step 1 to determine the final location of the mable
+3. Repeat steps 1-2 for each simulated marble
+4. Visualize the result
+
+Below is the simulation for a single marble moving down a 10-layered Galton board.
+Note how the integers 0 and 1 generated by `integers()` and converted to a
+series of -1 and +1 values, respectively.
+
+```{code-cell} ipython3
+moves = 2 * rng.integers(0, high=2, size=10) - 1
+final_position = moves.sum()
+final_position
+```
+
+To simulate additional marbles, we could use a `for` loop, but because we need
+the same number of integers for each marble, a more efficient way is to generate
+the random integers in a two-dimensional array with a row for every level on the
+Galton board and column for every marble simulated.
+We then sum up the values in each column to obtain the final positions of all marbles.
+
+```{code-cell} ipython3
+def galton(marbles, levels=10):
+    '''integer -> ndarray of integers
+
+    This function takes in an integer number of marbles and number of levels and outputs 
+    the horizontal position of these marbles at the bottom of a Galton board. 
+    '''
+    moves = 2 * rng.integers(0, high=2, size=(levels, marbles)) - 1
+    final_position = moves.sum(axis=0)
+    
+    return final_position
+```
+
+```{code-cell} ipython3
+sim = galton(1200)
+sim[:20]
+```
+
+We visualize the results below using a histogram plot.
+Being that all values are integers, we set `align='left'` so that each bar
+resides directly over the integer it represents.
+
+```{code-cell} ipython3
+bins = np.arange(-10,12)
+plt.hist(sim, bins=bins, align='left')
+plt.xticks(bins[::2])
+plt.xlabel('Horizontal Marble Position')
+plt.ylabel('Number of Marbles');
+```
+
+One interesting trend that appears in the above histogram is that all locations
+are integers.
+This is because we used whole numbers for our left and right movements of
+marbles (i.e., +1 and -1) and used ten layers in the above simulation.
+There is no way to produce an odd integer by totaling a series of ten +1 or -1
+integers.
+If we changed the number of layers to an odd number, we'd only get odd positions
+in the result, and if we used +1/2 and -1/2 for our horizontal movement, we'd get
+both even and odd integers.
```

Comment:
I think you can probably omit a lot of this explanation if you just left the values as 0s and 1s and just say that 0 means the left path was taken and 1 means the right path was taken.